### DISCUSSION TITLE
Using "consumer" to describe the current state of the line

### DISCUSSION THREAD

**Discussion by o-alquimista** [brgl/libgpiod#84] [General]
Is it ok to use "--consumer" to hold variable information on the current state of the line? e.g. `--consumer="my-script-current-power-50%"`

```sh
# gpioinfo -c gpiochip0 GPIO18      # my script would parse this
gpiochip0 18	"GPIO18"        	output consumer="my-script-current-power-50%"
```

This is assuming the line can have more than two states, so you cannot rely on the active/inactive description. In my case, I'm doing software-timed PWM.

I believe it should be ok to do that — it's just a string after all — but who knows, there could be a better way. My alternative would be managing a text file in `/var/run/`.

**Comment by brgl** [comment]
You can put whatever you want into the consumer string. How are you managing PWM with libgpiod is a mistery to me though. :)

**Reply by o-alquimista** [reply]
I'm using something like `--toggle 50ms,10ms` to get a slower fan speed.

**Reply by mfide** [reply]
just a reminder that GPIO_MAX_NAME_SIZE=32
